### PR TITLE
added lines of code to support deleting an existing address

### PR DESCRIPTION
**Pull request by azhelle16**
code to be reviewed
comments are to be removed once code is ready for prod

Name    :   Maricel Louise Sumulong
Date    :   April 19, 2020 PDT
E-mail  :   mllsumulong@gmail.com

### CODE REVIEW

File: app/controllers/change_requests_controller.rb [hunk 1]
```diff
@@ -180,17 +180,34 @@ def persist_change(change_request)
...
-      rescue => error
-        puts "google geocoding failed for address " + address.id.to_s + ": " + error.message
-      end
+      if action.downcase == "remove"
+        #address.destroy
+        #puts "ID: #{change_request.object_id.to_s}"
+        begin
+          address.destroy
+        rescue => error
+          puts error.message
```
Comment (richardxia) [inline app/controllers/change_requests_controller.rb:197]: Just wondering, were you encountering errors trying to call `#destroy` on the Address? If so, what was the error?

File: app/controllers/change_requests_controller.rb [hunk 2]
```diff
@@ -180,17 +180,34 @@ def persist_change(change_request)
       phone.update field_change_hash
     elsif change_request.is_a? AddressChangeRequest
       puts "AddressChangeRequest"
+      # action = change_request.field_changes["action"]
+      action = "" # to be removed for end-to-end testing
```
Comment (richardxia) [inline app/controllers/change_requests_controller.rb:184]: The `action` is an enum on the [`ChangeRequest`](https://github.com/ShelterTechSF/askdarcel-api/blob/3ad4773352ddd5d0c089008197a1035d1279733c/app/models/change_request.rb#L5) model, meaning that once it's saved to the database, it is represented as an integer. This means that you should be able to check the enum value directly on the model instance, and not the `field_change_hash`, using either the name of the enum property or the special boolean methods for each enum value like so:

```rb
change_request.status  # Gives you a string holding "insert", "edit", or "remove"
change_request.remove? # Returns true if and only if the status is set to "remove"
```

See https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Enum.html for some docs on how Rails enums work.

**Comment by richardxia** [comment]
Thanks for submitting this, this is a good start!

**Comment by katerina-kossler** [comment]
Stale Branch - working on a new branch for 1319 so closing this and another out

**Pull request closed without merging**